File: skills/migrate-statsig/test-fixtures/phase2-examples/python-server/src/confidence_client.py
"""AFTER - migrated Confidence setup (Python LOCAL-resolve / in-process).

Uses the Confidence local-resolve OpenFeature provider
(`confidence-openfeature-provider`, from spotify/confidence-resolver), which
evaluates locally via WASM. `set_provider_and_wait` blocks until the initial
resolver state is fetched, replacing Statsig's `statsig.initialize(secret)`.
"""

import os

from openfeature import api
from confidence import ConfidenceProvider

_client = None


def get_client():
    global _client
    if _client is None:
        provider = ConfidenceProvider(
            client_secret=os.environ["CONFIDENCE_FLAG_CLIENT_SECRET"]
        )
        api.set_provider_and_wait(provider)
        _client = api.get_client()
    return _client

File: skills/migrate-statsig/test-fixtures/phase2-examples/java-server/src/main/java/com/example/ConfidenceClient.java
// AFTER — migrated Confidence setup (Java LOCAL-resolve / in-process).
//
// Replaces Statsig's Statsig.initializeAsync(secret).get() readiness with the
// Confidence local-resolve provider. setProviderAndWait blocks until the initial
// resolver state is fetched, so the hand-rolled init wait is gone.

package com.example;

import com.spotify.confidence.sdk.OpenFeatureLocalResolveProvider;
import dev.openfeature.sdk.Client;
import dev.openfeature.sdk.OpenFeatureAPI;

public final class ConfidenceClient {

  private static Client client;

  private ConfidenceClient() {}

  public static synchronized Client get() {
    if (client == null) {
      OpenFeatureLocalResolveProvider provider =
          new OpenFeatureLocalResolveProvider(System.getenv("CONFIDENCE_FLAG_CLIENT_SECRET"));
      OpenFeatureAPI.getInstance().setProviderAndWait(provider);
      client = OpenFeatureAPI.getInstance().getClient();
    }
    return client;
  }
}
